pyomo/common/errors.py

```diff
@@ -25,3 +25,10 @@ def __str__(self):
                  % ( repr(self.parameter), ) )
 
 
+class InfeasibleConstraintException(Exception):
```

Comment:
As a basic design question, should we make a general class that all Pyomo exceptions inherit off of?  Something like:
```python
class PyomoException(Exception):
    pass
```

Reply:
@jsiirola I think this is a great suggestion. This would enable packages that wrap Pyomo to catch all pyomo exceptions. I will make this change.

Reply:
I am putting this in another PR which will be ready shortly.